vm/internal/backend/containers.go

```diff
@@ -36,11 +39,11 @@ func GetContainersForVolume(ctx context.Context, cli *client.Client, volumeName
 	return containerNames
 }
 
-func StopContainersAttachedToVolume(ctx context.Context, cli *client.Client, volumeName string) ([]string, error) {
+func StopRunningContainersAttachedToVolume(ctx context.Context, cli *client.Client, volumeName string) ([]string, error) {
 	var stoppedContainersByExtension []string
 	var timeout = 10 * time.Second
 
-	containerNames := GetContainersForVolume(ctx, cli, volumeName)
+	containerNames := GetContainersForVolume(ctx, cli, volumeName, filters.NewArgs(filters.Arg("status", "running")))
```

Comment:
nit: From [the list of status](https://docs.docker.com/engine/reference/commandline/ps/#status), I wonder if we should also filter "restarting" containers. When restarting, and during the backup process, they could also alter the data.
And could "removing" containers also add or remove data from an attached volume?

Reply:
Yeah, I had the same thought, there's a paused state as well, i am not sure how's the behaviour of the state transitions are, i can add these 3 as well in the filters, 

[Possible](https://docs.docker.com/engine/reference/commandline/ps) status:  created, restarting, running, removing, paused, exited, or dead

I think we can take into consideration these 4: restarting, running, removing, paused.
In case of restarting, we can start it just fine,
And in case of removing, we can wait it's removal? 
For paused,I'm not sure how we should handle it, although since it's paused, there shouldn't be a risk of data corruption, so we can probably ignore it.

Reply:
> In case of restarting, we can start it just fine.

I don't know if it is possible that a restarting container can write data into the volume. I guess yes, it could generate some logs that could be written into the volume.
So, when the backup is done, yes I agree, starting it seams fine. 
However, if there is a container that is restarting when starting the backup, it could add data to the attached volume. So the container must be stopped.

> And in case of removing, we can wait it's removal? 

Looks like a good idea.

> For paused,I'm not sure how we should handle it, although since it's paused, there shouldn't be a risk of data corruption, so we can probably ignore it.

I agree

Reply:
so I added changes to support restarting and removing statuses, 
for restarting, we just treat it as running, stop it before cloning, and start it after cloning completes.
for removing, wait for removal to complete before we finish the StopRunningContainersAttachedToVolume function